Header header
ItemMsg target_item
ItemMsg[] items_in_bin
